keps/sig-testing/2420-reducing-kubernetes-build-maintenance/README.md

```diff
@@ -0,0 +1,664 @@
+- Removing Bazel from sub-projects other than the core repo
+  - Bazel comes with tradeoffs, each subproject can make its own decision in this regard. This KEP **only** covers the main Kubernetes repository, and nothing else.
+- Improving the previously existing make build
+  - We should strongly consider improving the implementation and behavior of this build in the future, but this is largely orthogonal to whether we should consider maintaining two build systems.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+1. Switch remaining CI usage (mostly a few presubmits) to use the make build.
+  - Most of CI already uses the make builds, excluding some presubmits, we will need to switch these (generally a flag flip in the CI configuration).
+  - Most of periodic testing consumes pre-uploaded binaries from the make builds, and does not build at all. These will require no changes.
+  - In areas where the make build generates fewer artifacts or exercises fewer paths than bazel, we will err on the side of parity with artifacts that end up in a kubernetes/kubernetes release
+2. Remove the bazel build and associated tooling.
+  - There are multiple scripts and LOTS of files related wholly to the bazel build in Kubernetes. Once we are confident that CI is no longer reliant on them we can remove these and relieving the maintenance toil.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+Non-blocking “make build” equivalents to kubernetes/kubernetes “bazel” CI jobs will be introduced (if they don’t already exist).
+When the new jobs provide equivalent signal, they will be moved to blocking, and the old jobs will be retired.
+
+This is relevant for at least the following jobs:
+ - `pull-kubernetes-bazel-test` (this can be converted to ~ `make test`)
+ - `pull-kubernetes-bazel-test` (this largely overlaps with other presubmits, if not for testing ~`bazel build //...` and can  likely be removed)
```

Comment:
duplicate lines?

Reply:
this one should say `pull-kubernetes-bazel-build`
```suggestion
 - `pull-kubernetes-bazel-test` (this can be converted to ~ `make test`)
 - `pull-kubernetes-bazel-build` (this largely overlaps with other presubmits, if not for testing ~`bazel build //...` and can  likely be removed)
```

Reply:
this is fixed.